src/server/editorServices.ts

```diff
@@ -361,6 +361,10 @@ namespace ts.server {
         openRefCount = 0;
 
         constructor(public projectService: ProjectService, public projectOptions?: ProjectOptions) {
+            if(projectOptions && projectOptions.files){

```

Comment:
i do not think this is the right place to do this. the compiler should not just complain about js files in this list. similarly for modules.